src/components/backend-ai-resource-broker.ts

```diff
@@ -756,6 +758,20 @@ export default class BackendAiResourceBroker extends BackendAIPage {
     });
   }
 
+  _roundResourceDecimalPlaces(resourceSlots: Object, roundUpNumber = 2) {
+    Object.keys(resourceSlots).map((resource) => {
+      // convert undefined or NaN to 0
+      let resourceValue = Number(resourceSlots[resource] ?? 0).toString();
```

Comment:
If `resourceSlots[resource]` is `NaN`, it occurs `TypeError: Cannot read properties of undefined (reading 'length')`. 
```suggestion
      let resourceValue = Number(resourceSlots[resource] ? resourceSlots[resource] : 0).toString();
```